src/modbus/ModbusClient.cpp

```diff
@@ -40,11 +43,25 @@ bool ModbusClient::connect()
         return false;
     }
 
-    if (modbus_connect(m_modbus) == -1)
```

Comment:
Maybe we should move this reconnect logic outside modbus client so this call would not block